addons/crm_iap_lead_website/models/crm_reveal_rule.py

```diff
@@ -113,13 +113,15 @@ def _compute_lead_count(self):
     def action_get_lead_tree_view(self):
         action = self.env["ir.actions.actions"]._for_xml_id("crm.crm_lead_all_leads")
         action['domain'] = [('id', 'in', self.lead_ids.ids), ('type', '=', 'lead')]
-        action['context'] = dict(self._context, create=False)
+        filtered_context = self.with_context(group_by=[])
+        action['context'] = dict(filtered_context._context, create=False)
```

Comment:
The dict() signature is as follow: `dict(dict_to_copy_from, **key_values_to_update_the_copied_dict_with)`, you can pass as many kwargs as you want which is a much simplier than creating a new recordset with an updated environment.
```suggestion
        action['context'] = dict(self._context, create=False, group_by=[])
```
By the way, we try to get rid of `_context`, maybe we could take the opportunity to use `self.env.context` instead of `self._context`.

Note that it applies to most of your changes.